From 24c66ba31c1e137bda96cd88cda9b08391d0459b Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 21 Sep 2022 12:53:44 +0300
Subject: [PATCH] [SW-106171] gaudi3: tpm: add baud rate function for asic

spi baud rate required for tpm is hardcoded for onix platform. But
for asic the same needs to be calculated from pll . This patch
adds the implementation for the same.

Change-Id: I66508e2ad50c88790cb78cd63a0cd95db6166b48
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 6f478f0..6053709 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -15,3 +15,19 @@
 {
 	return get_pcie_pll_freq(PLL_DIV3);
 }
+
+#define HL_SPI_FLASH_MAX_FREQ		(SYS_REF_CLOCK / 2) /* 50MHz/2 = 25MHz*/
+
+uint64_t hl_spi_baudr_get(void)
+{
+	uint32_t factor;
+	uint64_t pcie_div3_freq;
+
+	pcie_div3_freq = get_pcie_pll_freq(PLL_DIV3);
+	factor = pcie_div3_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	/* baud rate doesn't support odd number and must not be 0 */
+	factor = (factor < 2) ? 2 : factor;
+
+	return (factor + 1) & 0xFFFE;
+}
